ui/components/component-library/avatar-token/avatar-token.js

```diff
@@ -0,0 +1,92 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { BaseAvatar } from '../base-avatar';
+
+import {
+  COLORS,
+  SIZES,
+  DISPLAY,
+  ALIGN_ITEMS,
+  JUSTIFY_CONTENT,
+} from '../../../helpers/constants/design-system';
+
+export const AvatarToken = ({
+  size = SIZES.MD,
+  backgroundColor = COLORS.BACKGROUND_ALTERNATIVE,
+  borderColor = COLORS.BORDER_DEFAULT,
+  tokenName,
+  tokenImageUrl,
+  showHalo,
+  ...props
+}) => {
+  const [showFallback, setShowFallback] = useState(!tokenImageUrl);
+
+  const baseStyle = showHalo ? { position: `relative` } : {};
+  const sizePercentage = showHalo ? '62.5%' : '100%';
+
```

Comment:
I have calculated `sizePercentage` based on Figma for tokenArt to background ratio.